### PR TITLE
update ARM metadata

### PR DESCRIPTION
**Pull request by dikhakha**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/Tests/L0.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe('AzureRmWebAppDeployment Suite', function() {
             assert(tr.stderr.length == 0 || tr.errorIssues.length, 'should not have written to stderr');
             var expectedOut = 'Updated history to kudu'; 
             assert(tr.stdout.search(expectedOut) > 0, 'should have said: ' + expectedOut);
-            expectedOut = 'Successfully updated scmType to VSTSRM';
+            expectedOut = 'loc_mock_SuccessfullyUpdatedAzureRMWebAppConfigDetails';
```
Comment (sachinma) [inline Tasks/AzureRmWebAppDeployment/Tests/L0.ts]: Where is this defined and how are you testing that the new message is getting printed to ensure that we called UpdateMetadat?

Comment (dikhakha) [inline Tasks/AzureRmWebAppDeployment/Tests/L0.ts]: loc_mock_SuccessfullyUpdatedAzureRMWebAppConfigDetails is printed only after both update operations (ScmType and Metadata) are successful.

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 2]
```diff
@@ -584,6 +584,65 @@ export async function getAppServiceDetails(endpoint, resourceGroupName: string,
     return deferred.promise;
 }
 
+export async function getAzureRMWebAppMetadata(endpoint, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string) {
```
Comment (dikhakha) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: will do.

File: Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts [hunk 3]
```diff
@@ -177,9 +177,8 @@ async function run() {
             var kuduWorkingDirectory = virtualApplication ? virtualApplicationPhysicalPath : 'site/wwwroot';
             await kuduUtility.runPostDeploymentScript(publishingProfile, kuduWorkingDirectory, scriptType, inlineScript, scriptPath, takeAppOfflineFlag);
         }
-        await updateScmType(endPoint, webAppName, resourceGroupName, deployToSlotFlag, slotName);
-
-        }
+    }
+    await updateWebAppConfigDetails(endPoint, webAppName, resourceGroupName, deployToSlotFlag, slotName);
```
Comment (dikhakha) [inline Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts]: will update config details for both Linux and windows app.

**Pull request merged**